samples/bluetooth/hci_uart/src/main.c

```diff
@@ -400,5 +402,25 @@ void main(void)
 		if (err) {
 			LOG_ERR("Failed to send");
 		}
+
+#if defined(CONFIG_INIT_STACKS)
+		static u32_t prio_ts;
+
+		if (k_uptime_get_32() - prio_ts > K_SECONDS(5)) {
+			STACK_ANALYZE("main thread stack",
+				      z_main_stack);
+			STACK_ANALYZE("ISR stack",
+				      _interrupt_stack);
+			STACK_ANALYZE("idle thread stack",
+				      z_idle_stack);
+			STACK_ANALYZE("sys workqueue stack",
+				      sys_work_q_stack);
+			prio_ts = k_uptime_get_32();
+		}
+
+		if (k_uptime_get_32() - prio_ts > K_SECONDS(5)) {
+			prio_ts = k_uptime_get_32();
+		}
```

Comment:
This whole branch seems unnecessary to me. The condition is identical to the branch above it and you're not doing any long-duration work in that branch.

Reply:
yeah, removed it.